SEED-platform/seed
# feat(docker): add docker for local development

#### Any background context you want to provide?
Currently, there's no convenient way start developing with docker.
#### What's this PR do?
- Makes it easy for developers to develop locally using docker containers, providing live reload, debugging, and testing.
- changes our docker-compose files from 3.0 to 3.4. This is a potentially breaking change for some servers, as 3.4 requires requires docker engine 17.09.0 or higher. I checked our BRICR server to see what it's running and it's safe - we still need to confirm our other servers are compatible
```bash
$ docker --version && docker-compose --version
Docker version 19.03.6, build 369ce74a3c
docker-compose version 1.25.4, build 8d51620a
```
#### How should this be manually tested?
Follow the instructions lined out in setup_docker.rst. You should be able to:
- start the server and navigate the site
- run the debugger
- run tests
- use a different database volume other than seed_pgdata
- customize server settings with local_untracked.py
